.github/workflows/build-test-verilator.yml

```diff
@@ -135,14 +113,93 @@ jobs:
           path: /opt/riscv
           key: riscv-${{ env.RISCV_VERSION }}-${{ env.SCCACHE_C_CUSTOM_CACHE_BUSTER }}
 
+      - name: Setup riscv path
+        run: |
+          echo /opt/riscv/bin >> $GITHUB_PATH
+
+  build_matrix:
+    name: Build Smoke Test matrix
+    runs-on: ubuntu-22.04
+    needs: build_tools
+    outputs:
+      test_names: ${{ steps.output-matrix.outputs.test_names  }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install deps
+        run: |
+          sudo apt-get update -qy && sudo apt-get install -qy --no-install-recommends \
+            python3-minimal python3-yaml
+      - name: Build matrix
+        id: output-matrix
+        run: |
+          echo "test_names=$(python3 .github/scripts/build_tests_matrix.py)" >> $GITHUB_OUTPUT
+
+  build_and_test:
+    name: Verilator Smoke Test
+    runs-on: ubuntu-22.04
+    needs: build_matrix
+
+    strategy:
+      fail-fast: false
+      matrix:
+        test_name: ${{fromJSON(needs.build_matrix.outputs.test_names)}}
+
+    env:
+      # TODO: is there a way to grab these from build_tools?
```

Comment:
This could be done via the job outputs feature (https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idoutputs).